### PR TITLE
[node] fix error on `http2.ClientSessionOptions` on typescript 5.1 with `exactOptionalPropertyTypes`

### PR DESCRIPTION
**Pull request by DetachHead**
fixes #64613

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes

### CODE REVIEW

**Comment by Semigradsky** [comment]
It looks like `createConnection` should not be in `SessionOptions`. For example I don't see `createConnection` in `createServer` options - https://nodejs.org/dist/latest-v18.x/docs/api/http2.html#http2createserveroptions-onrequesthandler

File: types/node/http2.d.ts [hunk 1]
```diff
@@ -1053,7 +1053,6 @@ declare module 'http2' {
          */
         unknownProtocolTimeout?: number | undefined;
         selectPadding?(frameLen: number, maxFrameLen: number): number;
-        createConnection?(authority: url.URL, option: SessionOptions): stream.Duplex;
```
Comment (Semigradsky) [inline types/node/http2.d.ts:1056]: `types/node/ts4.8/http2.d.ts` also.

 Make sense to fix for `v14`/`v16`/`v14/ts4.8`/`v16/ts4.8` too.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@DetachHead thanks!

**Comment by DetachHead** [comment]
Ready to merge

**Pull request merged**